pdns/ws-auth.cc

```diff
@@ -656,36 +646,42 @@ static void apiZoneCryptokeysPOST(DNSName zonename, HttpRequest *req, HttpRespon
     throw ApiException("Invalid keytype " + stringFromJson(document, "keytype"));
   }
 
-  UeberBackend B;
-  DNSSECKeeper dk(&B);
+  int64_t insertedId;
 
-  int insertedId;
   if (content.is_null()) {
-    int bits = intFromJson(document, "bits", 0);
+    int bits = 0;
+    auto docbits = document["bits"];
+    if (!docbits.is_null()) {
+      if (!docbits.is_number() || (fmod(docbits.number_value(), 1.0) != 0) || docbits.int_value() < 0) {
+        throw ApiException("'bits' must be a positive integer value");
+      } else {
+        bits = docbits.int_value();
+      }
+    }
```

Comment:
Because a wrong user input would have been changed to the value 0 without the user knowing.